BackHandler.android.js

```diff
@@ -8,6 +8,7 @@ class BackHandlerAndroid extends React.Component {
 
   constructor(props) {
     super(props);
+    if (props.navigation.isFocused()) BackHandler.addEventListener('hardwareBackPress', this.onBackPressed);
```

Comment:
if `props.navigation.isFocused()` then we probably don't need to add the 'willFocus' listener on the next line, right? Can you put that into `else` branch?

Reply:
At creation that would be ok, but it breaks for subsequent visits of the screen. E.g.:
 * Visit screen `A` -> subscribes to `hardwareBackPress `
 * Navigate to `B` -> `A` unsubscribes from the backhandler, but is still mounted
 * go back to `A` -> since there is no `willFocus` listener we do not re-subscribe to `hardwareBackPress`

I think both lines are needed

Reply:
ah, right